Model/Config/HealthCheck.php

```diff
@@ -0,0 +1,250 @@
+            ->setPlatform("Magento/".$this->_helper->getMagentoVersion()."Zip_ZipPayment/".$this->_helper->getExtensionVersion());
+
+
+        $curlEnabled = function_exists('curl_version');
+        $publicKey = $this->_config->getMerchantPublicKey();
+        $privateKey = $this->_config->getMerchantPrivateKey();
+
+        // check if private key is empty
+        if (empty($privateKey)) {
+            $this->appendItem(self::STATUS_ERROR, self::API_PRIVATE_KEY_INVALID_MESSAGE);
+        }
+
+        // check if public key is empty
+        if (empty($publicKey)) {
+            $this->appendItem(self::STATUS_ERROR, self::API_PUBLIC_KEY_INVALID_MESSAGE);
+        }
+
+        // check whether SSL is enabled
+        $this->checkStoreSSLSettings();
+
+        // check whether CURL is enabled ot not
+        if (!$curlEnabled) {
+            $this->appendItem(self::STATUS_ERROR, self::CURL_EXTENSION_DISABLED);
+        } else {
+            $curlObject->setConfig(
+                array(
+                    'timeout' => 10,
+                )
+            );
+
+            try {
+                $apiConfig->setCurlTimeout(30);
+                $headers = array(
+                    'Authorization: ' .
+                    $apiConfig->getApiKeyPrefix('Authorization') .
+                    ' ' .
+                    $apiConfig->getApiKey('Authorization'),
+                    'Accept : application/json',
+                    'Zip-Version: 2017-03-01',
+                    'Content-Type: application/json',
+                    'Idempotency-Key: ' .uniqid()
+                );
+                $url = $apiConfig->getHost().'/me';
+                $isAuEndpoint = false;
+
+                // check api key length if it is more than or equal 50 then call SMI merchant info endpoint
+                // otherwise call checkout get api endpoint only for Australia
+                if (strlen($privateKey) <= 50) {
+                    $checkoutId = 'au-co_PxSeQfLlpaYn6bLMZSMv13';
+                    $url = $apiConfig->getHost().'/checkouts/'.$checkoutId;
+                    $isAuEndpoint = true;
+                }
+                $curlObject->write(\Zend_Http_Client::GET, $url, '1.1', $headers);
+                $response = $curlObject->read();
+                $sslVerified = $curlObject->getInfo(CURLINFO_SSL_VERIFYRESULT) == 0;
+                $httpCode = $curlObject->getInfo(CURLINFO_HTTP_CODE);
+                // if API certification invalid
+                if (!$sslVerified) {
+                    $this->appendItem(self::STATUS_WARNING, self::API_CERTIFICATE_INVALID_MESSAGE);
+                }
+
+                // if API credential is invalid
+                if ($httpCode == '401') {
+                    $this->appendItem(self::STATUS_ERROR, self::API_CREDENTIAL_INVALID_MESSAGE);
+                }
+                if ($httpCode == '200' && $isAuEndpoint == false) {
```

Comment:
Response code from monolith could be 201, can we check any 2xx status instead of 200